typing/type_layout.ml

```diff
@@ -110,130 +77,80 @@ module Violation = struct
           format l1 format l2
 end
 
-let sort_var () = Var (ref None)
-
-let any = Any
-let any_sort () = Sort (sort_var ())
-let value = Sort Value
-let immediate = Immediate
-let immediate64 = Immediate64
-let void = Sort Void
-
-let rec sort_repr s =
-  match s with
-  | (Value | Void) -> s
-  | Var r -> begin
-      match !r with
-      | Some s -> sort_repr s
-      | None -> s
-    end
-
-let repr l =
-  match l with
-  | (Any | Immediate | Immediate64) -> l
-  | Sort s -> Sort (sort_repr s)
-
 let all_void layouts =
   Array.for_all (fun l ->
-    match repr l with
-    | Sort Void -> true
-    | (Any | Immediate | Immediate64
-      | Sort (Value | Var _)) -> false) layouts
+    match Layout.get l with
+    | Const Void -> true
+    | Const (Any | Immediate | Immediate64 | Value) | Var _ -> false)
+    layouts
 
+(* CR layouts: this function is suspect; it seems likely to reisenberg
+   that refactoring could get rid of it *)
```

Comment:
It's suspect because it's partial. Specifically, my hope is that places where this function is used can be refactored just to store sorts instead of layouts. That loses the immediate/value distinction, but (as you know) I'm dubious of the whole idea of sorts to begin with. :) The CR is not intended to be addressed before the first release, but my hope is that it will get addressed before we wrap up the overall project.